pkg/prometheus/agent/operator.go

```diff
@@ -1395,3 +1418,66 @@ func (c *Operator) getMissingPermissions(ctx context.Context) (map[string][]stri
 
 	return missingPermissions, nil
 }
+
+// checkCRDAuthorization checks if the operator has access to the specified CRD
+// It checks individual namespaces if an allowlist was provided, otherwise it checks cluster-wide.
+func (c *Operator) checkCRDAuthorization(ctx context.Context, verb string, group string, resource string) (bool, error) {
+	if c.config.Namespaces.AllowList != nil {
+		allowed := true
+		for ns := range c.config.Namespaces.AllowList {
+			ssar := &authv1.SelfSubjectAccessReview{
+				Spec: authv1.SelfSubjectAccessReviewSpec{
+					ResourceAttributes: &authv1.ResourceAttributes{
+						Verb:      verb,
+						Group:     group,
+						Resource:  resource,
+						Namespace: ns,
+					},
+				},
+			}
+			ssarResponse, err := c.kclient.AuthorizationV1().SelfSubjectAccessReviews().Create(ctx, ssar, metav1.CreateOptions{})
+			if err != nil {
+				return false, err
+			}
+			if !ssarResponse.Status.Allowed {
+				allowed = false
+				break
+			}
+		}
+
+		return allowed, nil
+	}
+
+	ssar := &authv1.SelfSubjectAccessReview{
+		Spec: authv1.SelfSubjectAccessReviewSpec{
+			ResourceAttributes: &authv1.ResourceAttributes{
+				Verb:     verb,
+				Group:    group,
+				Resource: resource,
+			},
+		},
+	}
+	ssarResponse, err := c.kclient.AuthorizationV1().SelfSubjectAccessReviews().Create(ctx, ssar, metav1.CreateOptions{})
+	if err != nil {
+		return false, err
+	}
+	return ssarResponse.Status.Allowed, nil
+}
+
+// isScrapeConfigInstalled returns true if the CRD is installed and if the operator is allowed to access it
+func (c *Operator) isScrapeConfigInstalled(ctx context.Context) (bool, error) {
+	crdInstalled, err := k8sutil.IsAPIGroupVersionResourceSupported(c.kclient.Discovery(), monitoringv1alpha1.SchemeGroupVersion.String(), monitoringv1alpha1.ScrapeConfigName)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to check if the API supports the ScrapeConfig CRD")
+	}
+	if !crdInstalled {
+		return false, nil
+	}
+
+	ok, err := c.checkCRDAuthorization(ctx, "*", "monitoring.coreos.com", "scrapeconfigs")
+	if err != nil {
+		return false, errors.Wrap(err, "failed to check authorization on the ScrapeConfig CRD")
+	}
+
+	return ok, nil
+}
```

Comment:
I'm really not happy about that repetition, open to suggestions as to where to put this logic

Reply:
Since it's shared code between the two what about `pkg/prometheus/operator.go`?

Reply:
I see [here](https://github.com/prometheus-operator/prometheus-operator/pull/5476#discussion_r1157388043) and [here](https://github.com/prometheus-operator/prometheus-operator/pull/5476#discussion_r1157403667) that @simonpasquier mentioned refactoring those in `k8sutil`.  

@slashpai should I wait for you to merge your PR and reuse those functions?

Reply:
yes. I am updating that today